features/0420-rich-schemas-common/README.md

```diff
@@ -0,0 +1,251 @@
+DID's method name (for example `did:sov`) allows to identify Rich Schema objects with equal content within different 
+data registries (ledgers).   
+
+### Referencing Rich Schema Objects
+- Any Rich Schema object is referenced by other Rich Schema objects by its ID (DID).
+- A Rich Schema object may reference a Rich Schema object from another ledger (as defined by DID's method name).
+
+### Relationship
+- A credential definition refers to a single mapping object
+- A mapping object refers to 1 or more schema objects.
+Each attribute in a schema may be included in the mapping one or more times (it is possible to encode a single attribute 
+in multiple ways). A mapping may map only a subset of the attributes of a schema.
+- A presentation definition refers to 1 or more schema and credential definition objects. A presentation definition may use only a
+subset of the attributes of a schema.  
+
+![](relationship-diagram.png)
+
+### How Rich Schema objects are stored in the Data Registry
+
+Any write request for Rich Schema object has the same fields:
+```
+'id': <Rich Schema object's ID>                # DID string 
+'content': <Rich Schema object as JSON-LD>     # JSON-serialized string
+'rs_name': <rich schema object name>           # string
+'rs_version': <rich schema object version>     # string
+'rs_type': <rich schema object type>           # integer
+'ver': <format version>                        # integer                              
+```
+- `id` is a unique ID (for example a DID with a id-string being base58 representation of the SHA2-256 hash of the `content` field)
+- The `content` field here contains a Rich Schema object in JSON-LD format (see [0119: Rich Schema Objects](https://github.com/hyperledger/indy-hipe/tree/master/text/0119-rich-schemas)).
+It's passed and stored as-is.
+The `content` field must be serialized in the canonical form. The canonicalization scheme we recommend is the IETF draft 
+ [JSON Canonicalization Scheme (JCS).](https://tools.ietf.org/id/draft-rundgren-json-canonicalization-scheme-16.html)
+- `metadata` contains additional fields which can be used for human-readable identification    
+- `ver` defines the version of the format. It defines what fields and metadata are there, how `id` is generated, what hash function is used there, etc. 
+- Author's and Endorser's DIDs are also passed as a common metadata fields for any Request. 
+
+
+### Querying Rich Schema objects from the Data Registry
+- Any Rich Schema object can be get from a Data Registry (Ledger) by its ID (DID).
+- It should be possible to get Rich Schema objects by metadata as well: `(rs_name, rs_version, rs_type)`.
+- Currently it's supposed that every Rich Schema object is queried individually, so it's up to clients and applications
+to get, query and cache all dependent Rich Schema objects.
+- We may support resolving a Rich Schema object by the DID as a DID DOC object in a standard way once DID DOC is supported in Aries.
```

Comment:
What is meant by "once DID DOC is supported in Aries"?

Reply:
I reformulated it a bit:
`once aries-vdr interface for DID resolving is defined and DID DOC specification is finalized.`
As I know, there is no defined interface for DID resolving in aries shared libraries (aries vdri) yet.
Also as I know DID DOC specification is not finalized and a subject to change. 

So, the proposal is to keep in a mind a possibility to resolve rich schema objects by DIDs as DID DOCs, but not require to implement it right now.